testing/acceptance_test.py

```diff
@@ -893,61 +895,64 @@ def test_has_plugin(self, request):
         assert request.config.pluginmanager.hasplugin("python")
 
 
+def fake_time(monkeypatch: MonkeyPatch) -> None:
+    """Monkeypatch time functions to make TestDurations not rely on actual time."""
+    import time
+
+    current_time = 1586202699.9859412
+
+    def sleep(seconds: float) -> None:
+        nonlocal current_time
+        current_time += seconds
+
+    monkeypatch.setattr(time, "sleep", sleep)
+    monkeypatch.setattr(_pytest.runner, "time", lambda: current_time)
+    monkeypatch.setattr(_pytest.runner, "perf_counter", lambda: current_time)
```

Comment:
Have you considered using e.g. https://github.com/spulec/freezegun ?
(I think it is fine probably to not have an extra (test) dependency for now, but I assume there are other places where manipulating time programmatically would be good)

Reply:
No, didn't know about it. Doesn't look like it fits the exact use case though.

In my own project I never call date/time functions directly, only through some passed in object, so that it may be faked/mocked easily. But it can't be done here really.

BTW, Initially I went for doing

```py
class FakeTime:
    """A fake `time` module for TestDurations to make it not rely on actual time."""

    def __init__(self) -> None:
        self.current_time = 1586202699.9859412

    def sleep(self, seconds: float) -> None:
        self.current_time += seconds

    def time(self) -> float:
        return self.current_time

    def perf_counter(self) -> float:
        return self.current_time
```

and then

```py
testdir.monkeypatch.setattr(_pytest.runner, "time", FakeTime())
```

and that worked, but runs afoul of the "issue 185" thing.